components/bl-spoiler-effect/src/index.js

```diff
@@ -0,0 +1,298 @@
+
+  const [isApplied, setIsApplied] = useState(applyEffect);
+
+  const handleOnClick = useCallback(() => {
+    setIsApplied(!isApplied);
+    onClick();
+  }, [isApplied]);
+
+  component.applySpoiler = () => setIsApplied(true);
+  component.removeSpoiler = () => setIsApplied(false);
+
+  const Effect = EffectViews[effect];
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div
+      className={ cn('bl-customComponent-spoiler-effect', classList) }
+      ref={ elRef }
+      style={ style }
+      onClick={ handleOnClick }
+      onMouseEnter={ onMouseEnter }
+      onMouseMove={ onMouseMove }
+      onMouseLeave={ onMouseLeave }>
+
+      <div
+        className={ cn('spoiler-effect-content', { blurred: isApplied }) }
+        style={{ filter: isApplied ? `url(#filter-${effect})` : 'none' }}>
+        { pods.spoilerEffectContent.render() }
+      </div>
+
+      { isApplied && (
+        <svg className="svg" style={{ opacity: effect === EffectTypes.MORPHOLOGY ? DefaultValues.OPACITY : undefined }}>
+          <Effect fill={ background }/>
+        </svg>
+      ) }
+    </div>
+  );
+}
+
+function BlurEffect({ fill }) {
+  return (
+    <>
+      <defs>
+        <filter id="filter-blur">
+          <feGaussianBlur stdDeviation="8 16" x="0%" y="0%" in="SourceGraphic" edgeMode="none" result="blur"/>
+          <feColorMatrix
+            type="matrix"
+            values="1 0 0 0 0
+                    0 1 0 0 0
+                    0 0 1 0 0
+                    0 0 0 1 0"
+            in="blur"
+            result="coloredBlur"
+          />
+        </filter>
+      </defs>
+      <rect width="100%" height="100%" fill={ fill } filter="url(#filter-blur)"/>
+    </>
+  );
+}
+
+function MorphologyEffect({ fill }) {
+  return (
+    <>
+      <defs>
+        <filter id="filter-morphology">
+          <feMorphology operator="dilate" radius="2 7" x="0%" y="0%" in="SourceGraphic" result="morphology2"/>
+        </filter>
+      </defs>
+      <rect width="100%" height="100%" fill={ fill } filter="url(#filter-morphology)"/>
+    </>
+  );
+}
+
+function MorphoseEffect({ fill }) {
+  return (
+    <>
+      <defs>
+        <filter id="filter-morphose">
+          <feGaussianBlur stdDeviation="8 20" x="0%" y="0%" in="SourceGraphic" edgeMode="none" result="blur">
+            <animate attributeName="stdDeviation" values="8 20; 20 8; 8 20" dur="2s" repeatCount="indefinite"/>
+          </feGaussianBlur>
+          <feColorMatrix
+            type="matrix"
+            values="1 0 0 0 0
+                    0 1 0 0 0
+                    0 0 1 0 0
+                    0 0 0 1 0"
+            in="blur"
+            result="coloredBlur"
+          />
+        </filter>
+      </defs>
+      <rect width="100%" height="100%" fill={ fill } filter="url(#filter-morphose)"/>
+    </>
+  );
+}
+
+function MusicEffect({ fill }) {
+  const [baseFrequency, setBaseFrequency] = useState(getRandomBaseFrequency());
```

Comment:
```suggestion
  const [baseFrequency, setBaseFrequency] = useState(() => getRandomBaseFrequency());
```

And tell me why it needed :)

Reply:
Set the starting position of baseFrequency for subsequent work with it.
Or do you mean to give a start value without calling the function?

Reply:
In addition should say, this recalculation is absolutely useless. Function is called, but result isn't applied.